cpp/src/arrow/python/type_traits.h

```diff
@@ -105,7 +105,7 @@ struct npy_traits<NPY_FLOAT32> {
   using TypeClass = FloatType;
   using BuilderClass = FloatBuilder;
 
-  static constexpr float na_sentinel = NAN;
+  static constexpr float na_sentinel = std::numeric_limits<float>::quiet_NaN();
```

Comment:
Can you explain why this change is necessary? The `NAN` macro is standard: https://en.cppreference.com/w/cpp/numeric/math/NAN

Reply:
Added a comment, the issue is that the Windows implementation of the macro is not fully compatible with clang's stictness.

Reply:
Thank you. Just: "division" (typo)